scripts/ci/libraries/_build_images.sh

```diff
@@ -519,8 +519,8 @@ function build_images::build_ci_image() {
         " >"${DETECTED_TERMINAL}"
         spinner::spin "${OUTPUT_LOG}" &
         SPIN_PID=$!
-        # shellcheck disable=SC2064
-        traps::add_trap "kill ${SPIN_PID} || true" INT TERM HUP EXIT
+        # shellcheck disable=SC2064,SC2016
+        traps::add_trap '$(kill '${SPIN_PID}' || true)' EXIT HUP INT TERM
```

Comment:
```suggestion
        traps::add_trap '$(kill "${SPIN_PID}" || true)' EXIT HUP INT TERM
```

Does it not need `"` for expansion?

Reply:
Example with single quote:

```
❯ a="x"
❯ echo '$a'
$a
```

Reply:
Nope. It is entirely correct. Look where the scope of ' starts/ends.

I want the SPIN_PID to be resolved at the time when the add_trap command line parameter gets evaluated. The solution you proposed will make it evaluated at the moment trap will be executed.

```
SPIN_PID=10
traps::add_trap '$(kill '${SPIN_PID}' || true)' EXIT HUP INT TERM
```
will be evaluated to 

```
traps::add_trap '$(kill 10 || true)' EXIT HUP INT TERM
```

Simply the ${SPIN_PID} is between the two '-delimite striings  

And this is exactly what I want.